tutorials/basic-topics/sorting/merge-sort.md

```diff
@@ -11,4 +11,531 @@ keywords:
   - merge sort
 ---
 
-<TutorialAuthors names="@TBC"/>
+<TutorialAuthors names="@Sreetama2001"/>
+
+## Overview
+
+Merge Sort works by recursively breaking down an array into multiple subarrays and then after comparing each of the subarrays it arranges them into ascending or descending order by value and merges them into a single sorted array. Here I will illustrate `merge sort`  Leetcode problems in the following Languages.
+
+- Python 
+- C++
+- Java
+  
+## Few concepts on merge sort
+
+Suppose we have an array of integers, 
+
+`6 5 3 1 8 7 2 4`
+
+Note: In ``Python`` it is called **list** and in ``C++`` it can be called either **array or vector** and in ``Java`` it is called a **ArrayList**.
+
+Then we can see that merge sort is performed in this way. 
+
+![image](https://miro.medium.com/max/600/1*opwN0BhtH4zvPF697fPlow.gif)
+
+## Algorithm 
+
+### Divide (Dividing is our first step)
+
+- Calculate the midpoint by checking if the left index is less than the right index, if yes divide the array. 
+- Now continue dividing the array until `left index < right index` becomes false, that is until the division is not possible.
+     
+### Conquer
+
+- After dividing the array into the smallest units, start merging the elements again by comparing them.
+
+- We need to compare and merge starting from the last splits or last smallest units. So **Recursion** needs to be done here.
+
+### Merge 
+
+- Since each half is already sorted so we need to just sort between 2 halves to **combine/merge** than to make a bigger sorted array.
+   
+## [Leetcode Problem 912 Sort an Array](https://leetcode.com/problems/sort-an-array/)
+
+**Problem Statement** 
+
+Given an array of integers nums, sort the array in ascending order and return it.
+
+You must solve the problem without using any built-in functions in O(nlog(n)) time complexity and with the smallest space complexity possible.
+
+**Test Cases**
+    
+*Test case 1* :
+     
+```
+Input: nums = [5,2,3,1]
+Output: [1,2,3,5]
+Explanation: 
+After sorting the array, the positions of some numbers are not changed (for example, 2 and 3), while the positions of other numbers are changed (for example, 1 and 5).
+        
+```
+**Constraints as given in the problem:**
+
+$1 <= nums.length <= 5 * 10^4
+-5 * 10^4 <= nums[i] <= 5 * 10^4$ 
+
+*Here `insertion sort`, `quick sort`, `selection sort` and all other sorts in $O(n^2)$ will give **TLE** i.e Time Limit Exceeded*
+
+## STL Approach  
+Note: Not preferred here in this problem, but still you will get an AC in Leetcode
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+
+```cpp
+void stableStlSort(vector<int>& nums){
+    
```

Comment:
You can remove this empty line here.

Reply:
empty lines  after braces should be removed in a method ? @vigneshshiv

Reply:
It's not after braces like that, when you have few lines after method open brace, you can leave a empty line. 
But for only one line, it's not required to do it. 
Just remove line 87 empty line - that's enough

Reply:
ohk done formatting! It's my first time wring a blog that's why I have so many errors! Sorry for the inconveniences